sphinx/ext/intersphinx.py

```diff
@@ -124,15 +124,110 @@ def split_lines(iter):
+        auth username not given
+      * (``str`` or ``NoneType``) -- basic auth password or ``None`` if basic
+        auth password not given
+
+    :rtype: ``tuple``
+    """
+    url_parts = parse.urlsplit(url)
+    username = url_parts.username
+    password = url_parts.password
+    frags = list(url_parts)
+    # swap out "user[:pass]@hostname" for "hostname"
+    frags[1] = url_parts.hostname
+    url = parse.urlunsplit(frags)
+    return (url, username, password)
+
+
+def _read_from_url(url):
+    """Reads data from *url* with an HTTP *GET*.
+
+    This function supports fetching from resources which use basic HTTP auth as
+    laid out by RFC1738 § 3.1. See § 5 for grammar definitions for URLs.
+
+    .. seealso:
+
+       https://www.ietf.org/rfc/rfc1738.txt
+
+    :param url: URL of an HTTP resource
+    :type url: ``str``
+
+    :return: data read from resource described by *url*
+    :rtype: ``file``-like object
+    """
+    url, username, password = _strip_basic_auth(url)
+    handler = request.BaseHandler()
+
+    if username is not None and password is not None:
+        # case: url contains basic auth creds
+        password_mgr = request.HTTPPasswordMgrWithDefaultRealm()
+        password_mgr.add_password(None, url, username, password)
+        handler = request.HTTPBasicAuthHandler(password_mgr)
+
+    opener = request.build_opener(handler)
+
+    return opener.open(url)
+
+
+def _get_safe_url(url):
+    """Gets version of *url* with basic auth passwords obscured. This function
+    returns results suitable for printing and logging.
+
+    E.g.: https://user:12345@example.com => https://user:********@example.com
+
+    .. note::
+
+       The number of astrisks is invariant in the length of the basic auth
+       password, so minimal information is leaked.
+
+    :param url: a url
+    :type url: ``str``
+
+    :return: *url* with password obscured
+    :rtype: ``str``
+    """
+    safe_url = url
+    url, username, _ = _strip_basic_auth(url)
+    if username is not None:
+        # case: url contained basic auth creds; obscure password
+        url_parts = parse.urlsplit(url)
+        safe_netloc = '{0}:********@{1}'.format(username, url_parts.hostname)
+        # replace original netloc w/ obscured version
+        frags = list(url_parts)
+        frags[1] = safe_netloc
+        safe_url = parse.urlunsplit(frags)
+
+    return safe_url
+
+
 def fetch_inventory(app, uri, inv):
     """Fetch, parse and return an intersphinx inventory file."""
     # both *uri* (base URI of the links to generate) and *inv* (actual
     # location of the inventory file) can be local or remote URIs
     localuri = uri.find('://') == -1
+    if localuri is False:
+        # case: inv URI points to remote resource; strip any existing auth
+        uri, _, _ = _strip_basic_auth(uri)
     join = localuri and path.join or posixpath.join
     try:
         if inv.find('://') != -1:
```

Comment:
this should actually reuse "localuri" (not a problem of your patch, just saw it)